actors/builtin/market/market_test.go

```diff
@@ -352,9 +354,289 @@ func TestPublishStorageDeals(t *testing.T) {
 		actor.addParticipantFunds(rt, client, deal2.ClientBalanceRequirement())
 		actor.addProviderFunds(rt, deal2.ProviderBalanceRequirement())
 		rt.SetEpoch(publishEpoch + 1)
-		deal2ID := actor.publishDeal(rt, deal2)
+		deal2ID := actor.publishDeals(rt, deal2)[0]
 		actor.activateDeals(rt, []abi.DealID{deal2ID}, endEpoch+1, provider)
 	})
+
+	t.Run("publish multiple deals for different clients and ensure balances are correct", func(t *testing.T) {
+		rt, actor := basicMarketSetup(t, marketActor, owner, provider, worker, client)
+		client1 := tutil.NewIDAddr(t, 900)
+		client2 := tutil.NewIDAddr(t, 901)
+		client3 := tutil.NewIDAddr(t, 902)
+
+		// generate first deal for
+		deal1 := generateDealProposal(client1, provider, abi.ChainEpoch(20), abi.ChainEpoch(50))
+		actor.addProviderFunds(rt, deal1.ProviderCollateral)
+		actor.addParticipantFunds(rt, client1, deal1.ClientBalanceRequirement())
+
+		// generate second deal
+		deal2 := generateDealProposal(client2, provider, abi.ChainEpoch(25), abi.ChainEpoch(60))
```

Comment:
Are the start/end epochs different for a reason? The differing numbers makes me think they are relevant to the test, but it appears not. If they're not, just use obviously-irrelevant numbers like `100`, 1000`, `42`, and the same for each deal.